src/layouts/bookmark/components/modal/folder-password.modal.tsx

```diff
@@ -0,0 +1,87 @@
+import { Button } from '@/components/button/button'
+import Modal from '@/components/modal'
+import { TextInput } from '@/components/text-input'
+import { useEffect, useRef, useState } from 'react'
+
+interface FolderPasswordModalProps {
+	isOpen: boolean
+	onClose: () => void
+	onConfirm: () => void
+	folderPassword: string
+}
+
+export function FolderPasswordModal({
+	isOpen,
+	onClose,
+	onConfirm,
+	folderPassword,
+}: FolderPasswordModalProps) {
+	const [password, setPassword] = useState('')
+	const [errorMessage, setErrorMessage] = useState('')
+	const inputRef = useRef<HTMLInputElement>(null)
+
+	useEffect(() => {
+		if (isOpen) {
+			setTimeout(() => {
+				inputRef.current?.focus()
+			}, 0)
+		}
+	}, [isOpen])
```

Comment:
This is good idea but when we are dealing with DOM its better to use `useLayoutEffect`. 

since its runs synchronously after all DOM mutations we may not need timeout. even if u are going to use timeout, consider the cleanup function.